apps/sms/services/test/unit/messaging/messaging_service_test.js

```diff
@@ -0,0 +1,628 @@
+      smsOpts = {
+        recipients: '123',
+        content: 'hola',
+        serviceId: 0 // we use 0 because it's falsy, to test it still works
+      };
+
+      MessagingService.mozMobileMessageClient.method
+        .withArgs('send').returns(Promise.resolve(successResult));
+      MessagingService.sendSMS(smsOpts).then((results) => {
+        sinon.assert.calledWithExactly(
+          MessagingService.mozMobileMessageClient.method,
+          'send', smsOpts.recipients, smsOpts.content, { serviceId: 0 }
+        );
+      }).then(done, done);
+
+    });
+
+    test('send with a serviceId in non-dual SIM setup', function(done) {
+      this.sinon.stub(Settings, 'hasSeveralSim').returns(false);
+      smsOpts = {
+        recipients: '123',
+        content: 'hola',
+        serviceId: 1
+      };
+
+      MessagingService.mozMobileMessageClient.method
+        .withArgs('send').returns(Promise.resolve(successResult));
+      MessagingService.sendSMS(smsOpts).then((results) => {
+        sinon.assert.calledWithExactly(
+          MessagingService.mozMobileMessageClient.method,
+          'send', smsOpts.recipients, smsOpts.content, undefined
+        );
+      }).then(done, done);
+
+    });
+
+    test('serviceId is a string containing a number', function(done) {
+      this.sinon.stub(Settings, 'hasSeveralSim').returns(true);
+      smsOpts = {
+        recipients: '123',
+        content: 'hola',
+        serviceId: '0'
+      };
+
+      MessagingService.mozMobileMessageClient.method
+        .withArgs('send').returns(Promise.resolve(successResult));
+      MessagingService.sendSMS(smsOpts).then((results) => {
+        sinon.assert.calledWithExactly(
+          MessagingService.mozMobileMessageClient.method,
+          'send', smsOpts.recipients, smsOpts.content, { serviceId: 0 }
+        );
+      }).then(done, done);
+
+    });
+
+    test('serviceId is a bad string', function(done) {
+      this.sinon.stub(Settings, 'hasSeveralSim').returns(true);
+      smsOpts = {
+        recipients: '123',
+        content: 'hola',
+        serviceId: 'oirutoirutoitr'
+      };
+
+      MessagingService.mozMobileMessageClient.method
+        .withArgs('send').returns(Promise.resolve(successResult));
+      MessagingService.sendSMS(smsOpts).then((results) => {
+        sinon.assert.calledWithExactly(
+          MessagingService.mozMobileMessageClient.method,
+          'send', smsOpts.recipients, smsOpts.content, undefined
+        );
+      }).then(done, done);
+    });
+  });
+
+  suite('sendMMS', function() {
+
+    setup(function() {
+      this.sinon.spy(SMIL, 'generate');
+    });
+
+    test('send to one recipient successfully', function(done) {
+      var mmsOpts = {
+        recipients: '123',
+        subject: null,
+        content: 'hola'
+      };
+
+      MessagingService.mozMobileMessageClient.method
+        .withArgs('sendMMS').returns(Promise.resolve(successResult));
+      MessagingService.sendMMS(mmsOpts).then((result) => {
```

Comment:
I think we need result in this test, but not necessary in the following tests for serviceId part in deed